File: firestore-https-java/src/main/java/com/example/xbill/mcp/https/Product.java
package com.example.xbill.mcp.https;

import java.util.Date;

/**
 * Represents a product in the inventory.
 *
 * @param id              the document ID
 * @param name            the name of the product
 * @param price           the price of the product
 * @param quantity        the quantity in stock
 * @param imgfile         the image filename
 * @param timestamp       the timestamp associated with the product
 * @param actualdateadded the actual date the product was added
 */
public record Product(
    String id,
    String name,
    double price,
    int quantity,
    String imgfile,
    Date timestamp,
    Date actualdateadded) {
}

File: firestore-https-kotlin/src/main/kotlin/com/example/mcp/server/model/Product.kt
package com.example.mcp.server.model

import kotlinx.serialization.Serializable

@Serializable
data class Product(
    val id: String? = null,
    val name: String,
    val price: Double,
    val quantity: Int,
    val imgfile: String,
    val timestamp: String,
    val actualdateadded: String,
)
